### PR TITLE
refactor: added helper function performSessionEnd

### PR DESCRIPTION
**Pull request by jaissica12**
## Background
- Session timeout calculations were duplicated in `initialize()` and `endSession()` methods
- Both methods checked for session timeout but used different approaches:
   - initialize() used Date object comparison
   - endSession() used millisecond arithmetic with inverted logic
- Session end operations (logging event, clearing data, nullifying session) were scattered inline within endSession()

## What Has Changed
- Added `performSessionEnd` helper - Extracted session end operations into a dedicated function
- Added `hasSessionTimedOut` helper - Consolidates duplicate session timeout calculation logic

## Screenshots/Video
- {Include any screenshots or video demonstrating the new feature or fix, if applicable}

## Checklist
- [X] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [X] I have added tests that prove my fix is effective or that my feature works.
- [X] I have tested this locally.

## Additional Notes
- The original code used different comparison operators:
- `initialize()` used `>` - session expired after 30:00.000
- `endSession()` used `>=` - session expired at 30:00.000

**As of now I have standardized on `>=`, let me know if you think otherwise**

## Reference Issue (For employees only. Ignore if you are an outside contributor)
- Closes https://go/j/SDKE-376

### CODE REVIEW

File: src/sessionManager.ts [hunk 1]
```diff
@@ -33,16 +33,7 @@ export default function SessionManager(
...
-                mpInstance._Store.SDKConfig.sessionTimeout * 60000;
-
-            if (
-                new Date() >
-                new Date(
-                    mpInstance._Store.dateLastEventSent.getTime() +
-                        sessionTimeoutInMilliseconds
-                )
-            ) {
+            if (hasSessionTimedOut(mpInstance._Store.dateLastEventSent.getTime())) {
```
Comment (alexs-mparticle) [inline src/sessionManager.ts]: If you follow my suggestion below, this can be destructured and cleaned up.

```suggestion

            const { dateLastEventSent, SDKConfig } = mpInstance._Store;
            const { sessionTimeout } = SDKConfig;
            if (hasSessionTimedOut(dateLastEventSent.getTime(), sessionTimeout)) {
```

File: src/sessionManager.ts [hunk 2]
```diff
@@ -234,4 +206,34 @@ export default function SessionManager(
...
+     * @returns true if the session has expired, false otherwise
+     */
+    function hasSessionTimedOut(lastEventTimestamp: number): boolean {
+        const sessionTimeoutInMilliseconds: number =
+            mpInstance._Store.SDKConfig.sessionTimeout * 60000;
+        const timeSinceLastEvent: number =
+            new Date().getTime() - lastEventTimestamp;
+
+        return timeSinceLastEvent >= sessionTimeoutInMilliseconds;
+    }
```
[the remark this answers is not in the document]
Comment (jaissica12) [reply src/sessionManager.ts]: Updated!

File: src/sessionManager.ts [hunk 3]
```diff
@@ -234,4 +206,34 @@ export default function SessionManager(
...
+     * - Logs a SessionEnd event
+     * - Clears session start date
+     * - Nullifies the session ID and related data
+     * - Resets the time-on-site timer
+     */
+    function performSessionEnd(): void {
+        mpInstance._Events.logEvent({
+            messageType: Types.MessageType.SessionEnd,
+        });
+        mpInstance._Store.sessionStartDate = null;
```
[the remark this answers is not in the document]
Comment (jaissica12) [reply src/sessionManager.ts]: Added `this.sessionStartDate = null;` to `Store. nullifySession()`

File: src/sessionManager.ts [hunk 4]
```diff
@@ -234,4 +210,31 @@ export default function SessionManager(
...
         }
     };
+
+    /**
+     * Checks if the session has expired based on the last event timestamp
+     * @param lastEventTimestamp - Unix timestamp in milliseconds of the last event
+     * @param sessionTimeout - Session timeout in minutes
+     * @returns true if the session has expired, false otherwise
+     */
+    function hasSessionTimedOut(lastEventTimestamp: number, sessionTimeout: number): boolean {
```
Comment (alexs-mparticle) [inline src/sessionManager.ts]: This might be an edge case, since we `lastEventTimestamp` and `sessionTimeout` have default values, but I would add a guard clause here.

File: test/src/tests-session-manager.ts [hunk 5]
```diff
@@ -293,6 +293,188 @@ describe('SessionManager', () => {
...
+                expect(mpInstance._Store.sessionId).to.not.equal('TEST-ID');
+                
+                // Test with persistence (via endSession)
+                const newSessionId = mpInstance._Store.sessionId;
+                sinon.stub(mpInstance._Persistence, 'getPersistence').returns({
+                    gs: {
+                        les: thirtyOneMinutesAgo.getTime(),
+                        sid: newSessionId,
+                    },
+                });
```
[the remark this answers is not in the document]
Comment (jaissica12) [reply test/src/tests-session-manager.ts]: Added

File: test/src/tests-session-manager.ts [hunk 6]
```diff
@@ -293,6 +293,188 @@ describe('SessionManager', () => {
...
+                
+                mParticle.init(apiKey, window.mParticle.config);
+                const mpInstance = mParticle.getInstance();
+                
+                sinon.stub(mpInstance._Persistence, 'getPersistence').returns({
+                    gs: {
+                        les: exactlyThirtyMinutesAgo.getTime(),
+                        sid: 'TEST-ID',
+                    },
+                });
```
[the remark this answers is not in the document]
Comment (jaissica12) [reply test/src/tests-session-manager.ts]: Added

**Review by rmi22186 — APPROVED** [review]

**Review by alexs-mparticle — APPROVED** [review]
